python/tests/io/test_fsspec.py

```diff
@@ -204,6 +204,191 @@ def test_writing_avro_file(generated_manifest_entry_file: Generator[str, None, N
+    output_file = adlfs_fsspec_fileio.new_output(location=f"abfss://tests/{filename}")
+    with output_file.create() as f:
+        f.write(b"foo")
+
+    input_file = adlfs_fsspec_fileio.new_input(f"abfss://tests/{filename}")
+    assert input_file.open().read() == b"foo"
+
+    adlfs_fsspec_fileio.delete(input_file)
+
+
+@pytest.mark.adlfs
+def test_fsspec_getting_length_of_file_adlfs(adlfs_fsspec_fileio: FsspecFileIO) -> None:
+    """Test getting the length of an FsspecInputFile and FsspecOutputFile"""
+    filename = str(uuid.uuid4())
+
+    output_file = adlfs_fsspec_fileio.new_output(location=f"abfss://tests/{filename}")
+    with output_file.create() as f:
+        f.write(b"foobar")
+
+    assert len(output_file) == 6
+
+    input_file = adlfs_fsspec_fileio.new_input(location=f"abfss://tests/{filename}")
+    assert len(input_file) == 6
+
+    adlfs_fsspec_fileio.delete(output_file)
+
+
+@pytest.mark.adlfs
+def test_fsspec_file_tell_adlfs(adlfs_fsspec_fileio: FsspecFileIO) -> None:
+    """Test finding cursor position for an fsspec file-io file"""
+
+    filename = str(uuid.uuid4())
+
+    output_file = adlfs_fsspec_fileio.new_output(location=f"abfss://tests/{filename}")
+    with output_file.create() as write_file:
+        write_file.write(b"foobar")
+
+    input_file = adlfs_fsspec_fileio.new_input(location=f"abfss://tests/{filename}")
+    f = input_file.open()
+
+    f.seek(0)
+    assert f.tell() == 0
+    f.seek(1)
+    assert f.tell() == 1
+    f.seek(3)
+    assert f.tell() == 3
+    f.seek(0)
+    assert f.tell() == 0
+
+    adlfs_fsspec_fileio.delete(f"abfss://tests/{filename}")
+
+
+@pytest.mark.adlfs
+def test_fsspec_read_specified_bytes_for_file_adlfs(adlfs_fsspec_fileio: FsspecFileIO) -> None:
+    """Test reading a specified number of bytes from an fsspec file-io file"""
+
+    filename = str(uuid.uuid4())
+    output_file = adlfs_fsspec_fileio.new_output(location=f"abfss://tests/{filename}")
+    with output_file.create() as write_file:
+        write_file.write(b"foo")
+
+    input_file = adlfs_fsspec_fileio.new_input(location=f"abfss://tests/{filename}")
+    f = input_file.open()
+
+    f.seek(0)
+    assert b"f" == f.read(1)
+    f.seek(0)
+    assert b"fo" == f.read(2)
+    f.seek(1)
+    assert b"o" == f.read(1)
+    f.seek(1)
+    assert b"oo" == f.read(2)
+    f.seek(0)
+    assert b"foo" == f.read(999)  # test reading amount larger than entire content length
+
+    adlfs_fsspec_fileio.delete(input_file)
+
+
+@pytest.mark.adlfs
+def test_fsspec_raise_on_opening_file_not_found_adlfs(adlfs_fsspec_fileio: FsspecFileIO) -> None:
+    """Test that an fsspec input file raises appropriately when the adlfs file is not found"""
+
+    filename = str(uuid.uuid4())
+    input_file = adlfs_fsspec_fileio.new_input(location=f"abfss://tests/{filename}")
+    with pytest.raises(FileNotFoundError):
+        input_file.open().read()
+
+    # filename is not propagated in FileNotFoundError for adlfs
```

Comment:
All those test cases are "identical" to S3 except this one.  The raised error does not contain a filename, so the same assert cannot be done.  Speaking of identical tests, would it make sense to parameterize them?

Reply:
Can we add the filename in the `open` or `read` call by catching `FileNotFoundError` and wrapping it? That seems worth an check in exception handling:

```python
def open(self):
    try:
        # original method contents
    except FileNotFoundError as e:
        if str(e) contains self.location:
            raise e
        else:
            raise FileNotFoundError(f"No such file: {self.location}") from e
```

Reply:
Makes sense to me, having a consistent error feedback is valuable.  See changes in fsspec.py -> `FsspecInputFile.open()`.
I might also open a PR upstream to [FsSpec ](https://github.com/fsspec/filesystem_spec)to have them change those errors at the root.